homeassistant/components/frontend/__init__.py

```diff
@@ -345,14 +366,39 @@ def update_theme_and_fire_event():
             )
         hass.bus.async_fire(EVENT_THEMES_UPDATED)
 
-    @callback
-    def set_theme(call):
+    async def save_theme_store():
+        """Update store with new theme data."""
+        store = hass.data[DATA_THEMES_STORE]
+        data = {
+            DATA_DEFAULT_THEME: hass.data[DATA_DEFAULT_THEME],
+            DATA_DEFAULT_DARK_THEME: hass.data.get(DATA_DEFAULT_DARK_THEME),
+        }
+        await store.async_save(data)
+
+    async def set_theme(call):
         """Set backend-preferred theme."""
         data = call.data
         name = data[CONF_NAME]
         if name == DEFAULT_THEME or name in hass.data[DATA_THEMES]:
             _LOGGER.info("Theme %s set as default", name)
             hass.data[DATA_DEFAULT_THEME] = name
+            await save_theme_store()
+            update_theme_and_fire_event()
+        else:
+            _LOGGER.warning("Theme %s is not defined", name)
```

Comment:
I think we should still save it?

Reply:
Why would we save it, probably a typo by the user?

Reply:
I guess you're right.